src/library-authoring/library-sidebar/LibrarySidebar.tsx

```diff
@@ -53,30 +59,45 @@ const LibrarySidebar = ({ library, collection }: LibrarySidebarProps) => {
     [SidebarBodyComponentId.ComponentInfo]: (
       currentComponentUsageKey && <ComponentInfoHeader library={library} usageKey={currentComponentUsageKey} />
     ),
-    [SidebarBodyComponentId.CollectionInfo]: <CollectionInfoHeader collection={collection} />,
+    [SidebarBodyComponentId.CollectionInfo]: (
+      currentCollectionHit && <CollectionInfoHeader library={library} collection={currentCollectionHit} />
+    ),
     unknown: null,
   };
 
   const buildBody = () : React.ReactNode => bodyComponentMap[sidebarBodyComponent || 'unknown'];
   const buildHeader = (): React.ReactNode => headerComponentMap[sidebarBodyComponent || 'unknown'];
 
+  const collectionQuery: SearchParams | undefined = currentCollectionHit ? {
+    filter: ['type = "collection"', `context_key = "${library.id}"`, `block_id = "${currentCollectionHit.blockId}"`],
+    limit: 1,
+  } : undefined;
+
   return (
-    <Stack gap={4} className="p-3 text-primary-700">
-      <Stack direction="horizontal" className="d-flex justify-content-between">
-        {buildHeader()}
-        <IconButton
-          className="mt-1"
-          src={Close}
-          iconAs={Icon}
-          alt={intl.formatMessage(messages.closeButtonAlt)}
-          onClick={closeLibrarySidebar}
-          size="inline"
-        />
+    <SearchContextProvider
+      extraFilter={[
+        `context_key = "${library.id}"`,
+        ...(currentCollectionHit ? [`collections.key = "${currentCollectionHit.blockId}"`] : []),
+      ]}
+      overrideQueries={{ ...(collectionQuery ? { collections: collectionQuery } : {}) }}
+    >
```

Comment:
We need a new provider in the sidebar while showing the LibraryHome. We need to refactor this in the future.

Reply:
Could you put this info in a `TODO` code comment please, so we don't lose track of it?